### PR TITLE
Added lenet architecture

### PR DESCRIPTION
**Pull request by iArunava**
- [x] Implement the architecture
- [x] Add tests
- [x] Add a pretrained model on MNIST

### CODE REVIEW

File: scratchai/learners/metrics.py [hunk 1]
```diff
@@ -45,15 +45,41 @@ def miou(pred, gt, nc, c2n=None):
   if isinstance(gt, torch.Tensor):
     gt = gt.detach().clone().cpu().numpy()
```
Comment (iArunava) [inline scratchai/learners/metrics.py]: `.clone()` and then `.detach()` just to be safe.

File: scratchai/learners/metrics.py [hunk 2]
```diff
@@ -45,15 +45,41 @@ def miou(pred, gt, nc, c2n=None):
...
   
-  iou = {}
-  miou = 0
-  for cls in range(nc):
-    inter = np.logical_and(pred == cls,  gt == cls).sum()
-    union = np.logical_or(pred == cls, gt == cls).sum()
-    iou[cls] = inter / union if inter != 0 and union != 0 else 1.
-    miou += iou[cls]
+  with torch.no_grad():
```
Comment (iArunava) [inline scratchai/learners/metrics.py]: Wrap the above lines with `torch.no_grad()`

File: scratchai/one_call.py [hunk 3]
```diff
@@ -38,16 +40,26 @@ def classify(path:str):
   a resnet output the predicted value.
```
Comment (iArunava) [inline scratchai/one_call.py]: Add a note that if `trf is None` then it defaults to `rz256_cc224_tt_normimgnet`

File: scratchai/one_call.py [hunk 4]
```diff
@@ -38,16 +40,26 @@ def classify(path:str):
...
+  trf = imgutils.get_trf('rz256_cc224_tt_normimgnet' if not trf else trf)
+
+  # Special Case: if net == 'lenet_mnist' then image needs to have one channel
+  if nstr == 'lenet_mnist': img = trf(imgutils.gray(path)).unsqueeze(0)
+  # Normal Cases
+  else: img = trf(imgutils.load_img(path)).unsqueeze(0)
+
+  # Edge Case: In case of 2D Images, the above line adds the channel dim
+  # And then the batch dim needs to be added.
+  if len(img) == 3: img = img.unsqueeze(0)
```
Comment (iArunava) [inline scratchai/one_call.py]: `img.shape()`

File: tests/test_nets.py [hunk 5]
```diff
@@ -144,3 +153,23 @@ def test_itnst(self):
...
+
+class TestCommon(nn.Module):
+  
+  def test_flatten(self):
+    n1 = torch.randn(13, 3, 4, 2)
+    out = nets.Flatten()(n1)
+    self.assertTrue(out.shape, [13, 3*4*2], 'out shape not okay')
+
+  def test_debug(self):
+    is_imp = getattr(nets, 'debug')
```
Comment (iArunava) [inline tests/test_nets.py]: Change this with `utils.check_if_implemented`

File: scratchai/learners/metrics.py [hunk 6]
```diff
@@ -45,15 +45,41 @@ def miou(pred, gt, nc, c2n=None):
...
+      iou[cls] = inter / union if inter != 0 and union != 0 else 1.
+      miou += iou[cls]
   
   # TODO print the iou in table format
   #print (tabulate([list(iou.keys()), list(iou.values())]))
 
   return miou/nc
+
+
+def accuracy(pred, gt, k=1):
```
Comment (iArunava) [inline scratchai/learners/metrics.py]: Accuracy calculated here: https://github.com/iArunava/scratchai/blob/e2fdc8f47232a18231975e176749c9ab4e9dcfe6/scratchai/learners/clflearner.py#L64-L67

Put the code in the above lines in this function, later this function can be made more general to return top-1 and top-5 accuracy.
